config/registry_image_pruner/image_pruner/test_prune_images.py

```diff
@@ -142,6 +142,23 @@ def test_remove_orphan_tags_with_expected_suffixes(self, urlopen):
                     "name": "sha256-071c766795a0.src",
                     "manifest_digest": "sha256:0ab207f62413",
                 },
+
+                # single deprecated source image tag remains
+                {"name": "123abcd.src", "manifest_digest": "sha256:1234566"},
+
+                # old image has only the deprecated source image, which should not be removed
+                {"name": "donotdelete", "manifest_digest": "sha256:1234567"},
+                {"name": "donotdelete.src", "manifest_digest": "sha256:1234568"},
+
+                # the binary image was deleted before. Now, these should be deleted.
+                {"name": "build-100.src", "manifest_digest": "sha256:1345678"},
+                {"name": "sha256-4567890.src", "manifest_digest": "sha256:1345678"},
```

Comment:
I don't quite understand this test case. Was the source tag supposed to be `sha256-1345678.src`?

Reply:
This aims to be a status that is the binary image `xxx:build-100` has been removed and the pruner starts to clean up.

Reply:
Ah right, I missed the **.src** in `build-100.src` :+1: